server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java

```diff
@@ -79,6 +93,20 @@ public class ShardsBatchGatewayAllocator implements ExistingShardsAllocator {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<TimeValue> PRIMARY_BATCH_ALLOCATOR_TIMEOUT_SETTING = Setting.timeSetting(
+        PRIMARY_BATCH_ALLOCATOR_TIMEOUT_SETTING_KEY,
+        TimeValue.MINUS_ONE,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    public static final Setting<TimeValue> REPLICA_BATCH_ALLOCATOR_TIMEOUT_SETTING = Setting.timeSetting(
+        REPLICA_BATCH_ALLOCATOR_TIMEOUT_SETTING_KEY,
+        TimeValue.MINUS_ONE,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
```

Comment:
min value for these setting so that very low values are not specified like 30sec

Reply:
Added min timeout of 20 seconds

Reply:
@imRishN this has not been updated

Reply:
I made the change but then reverted. We would need a way to disable timeouts which will be handled by `MINUS_ONE`, and min timeout of 20 sec wouldn't allow to have that. Hence keeping it like this. Do you see a major concern here?

Reply:
Opened tracking issue - https://github.com/opensearch-project/OpenSearch/issues/14941